xcm-support/src/lib.rs

```diff
@@ -75,3 +78,79 @@ impl UnknownAsset for () {
 		Err(DispatchError::Other(NO_UNKNOWN_ASSET_IMPL))
 	}
 }
+
+/// Extracts the `AccountId32` from the passed `location` if the network
+/// matches.
+pub struct RelaychainAccountId32Aliases<Network, AccountId>(PhantomData<(Network, AccountId)>);
+impl<Network: Get<NetworkId>, AccountId: From<[u8; 32]> + Into<[u8; 32]> + Clone>
+	xcm_executor::traits::Convert<MultiLocation, AccountId> for RelaychainAccountId32Aliases<Network, AccountId>
+{
+	fn convert(location: MultiLocation) -> Result<AccountId, MultiLocation> {
+		let id = match location {
+			MultiLocation {
+				parents: 1,
+				interior: X1(AccountId32 {
+					id,
+					network: NetworkId::Any,
+				}),
+			} => id,
+			_ => return Err(location),
+		};
+		Ok(id.into())
+	}
+
+	fn reverse(who: AccountId) -> Result<MultiLocation, AccountId> {
+		Ok(AccountId32 {
+			id: who.into(),
+			network: Network::get(),
+		}
+		.into())
```

Comment:
This will reverse to `MultiLocation {
				parents: 0,
				interior: X1(AccountId32),
}`, no?
That means it will not convert back to the input if calling `RelaychainAccountId32Aliases.convert(location).reverse()` which would be desirable, no?

Reply:
It's a good point, as the reverse path is not currently go through, I may miss here.

Reply:
after I change here to return `(1, AccountId32).into()`. If we config `(...,AccountId32Aliases,RelaychainAccountId32Aliases)`, then reverse on AccountId will always return `(parents:0, X1(AccountId32))`. But when I change the order to `(...,RelaychainAccountId32Aliases, AccountId32Aliases)`, then recerse on AccountId will always return `(parents:1, X1(AccountId32))`. this sounds no better solution.

Reply:
I've post an issue on polkadot: https://github.com/paritytech/polkadot/issues/4296. would you mind to see if that's a potential problem? @KiChjang @apopiak